test/utility.cpp

```diff
@@ -91,6 +91,60 @@ TEST_CASE("exchange", "[utility.exchange]") {
   }
 }
 
+// [utility.as_const], as_const
+
+// template<class T>
+//   constexpr add_const_t<T>& as_const(T& t) noexcept;
+// template<class T>
+//   void as_const(const T&&) = delete;
+namespace as_const_deleted {
+void as_const(...);
+
+namespace check {
```

Comment:
What's the purpose of the `check` namespace here?

Reply:
Turns out is not actually needed, and we could also remove the other one too. Should we?

Reply:
If by "the other one" you mean `as_const_deleted`, I'm fine with that one. The `no_result` inside of it is indeed a `as_const_deleted`-specific `no_result`

Reply:
Alright. I'll leave it, and we can still take it out if we ever decide to split utility tests in separate files.